crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -427,4 +544,32 @@ pub mod error {
         #[error("invalid solver competition data fetched from database: {0}")]
         InvalidSolverCompetition(anyhow::Error),
     }
+
+    #[derive(Debug, thiserror::Error)]
+    pub enum Settlement {
+        #[error("failed communication with the database: {0}")]
+        DatabaseError(sqlx::Error),
+        #[error("invalid settlement event")]
+        InvalidEvent,
+    }
+
+    impl From<DatabaseError> for Settlement {
+        fn from(err: DatabaseError) -> Self {
+            Self::DatabaseError(err.0)
+        }
+    }
+
+    #[derive(Debug, thiserror::Error)]
+    pub enum SettlementEvent {
```

Comment:
What's the difference semantically between `error::Settlement` and `error::SettlementEvent. If there is a good reason to keep them separate can we please document the types so that someone that sees this in the future understands what we thought when we added all these error types.

Reply:
deleted